satellite/audit/service2.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
+	"storj.io/storj/satellite/metainfo"
+)
+
+// Service2 contains information for populating audit queue and processing audits.
+type Service2 struct {
+	log *zap.Logger
+
+	reservoirChore *ReservoirChore
+	workers        []*worker
+	queue          *queue
+}
+
+// NewService2 instantiates Service2, ReservoirChore and workers.
+func NewService2(log *zap.Logger, config Config, metaloop *metainfo.Loop) (*Service2, error) {
+	queue := newQueue()
+	var workers []*worker
+	for i := 0; i < config.WorkerCount; i++ {
+		workers = append(workers, newWorker(queue))
+	}
+	return &Service2{
+		log: log,
+
+		reservoirChore: NewReservoirChore(log.Named("reservoir chore"), queue, metaloop, config),
```

Comment:
Services should not import chores. https://github.com/storj/docs/blob/master/code/Architecture.md#interactions-between-layers